src/Common/LRUCache.h

```diff
@@ -306,17 +377,18 @@ class LRUCache
         cell.size = cell.value ? weight_function(*cell.value) : 0;
         current_size += cell.size;
 
-        removeOverflow();
+        return true;
```

Comment:
Looks incorrect. In this case if there already was a value with the same key in `cells` (case `!inserted`), then when we update its value, it can have a different size (because we change previous value to `Mapped`), so there can be overflow, but now there is no `removeOverflow` at the end of this method.

Reply:
updated